etl/steps/data/garden/owid/2022-12-19/gdp_last_2000_years.py

```diff
@@ -0,0 +1,139 @@
+"""
+This code combines Maddison Project Database 2020 with Maddison Database 2010 and WDI to get historical and up-to-date GDP and GDP per capita data, using the WDI values from 1990 onwards and estimating
+GDP growth from Maddison data to retroactively adjust WDI data. Maddison Database 2010 is obtained for the 1 to 1820 GDP global estimations. These data is dropped in subsequent versions (Maddison Project Database, after Angus Maddison's death).
+
+"""
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import Names
+from etl.paths import DATA_DIR
+
+VERSION_MADDISON = "2020-10-01"
+VERSION_WDI = "2022-05-26"
+VERSION_MADDISON2010 = "2022-12-23"
+
+log = get_logger()
+
+# naming conventions
+N = Names(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("gdp_last_2000_years.start")
+
+    # Load Maddison garden step
+    ds = Dataset(DATA_DIR / f"garden/ggdc/{VERSION_MADDISON}/ggdc_maddison")
+    df_maddison = ds["maddison_gdp"]
+    df_maddison = pd.DataFrame(df_maddison, dtype=np.float64)
+    df_maddison = df_maddison.reset_index()
+    df_maddison = df_maddison.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Load WDI garden step
+    ds = Dataset(DATA_DIR / f"garden/worldbank_wdi/{VERSION_WDI}/wdi")
+    df_wdi = ds["wdi"]
+    df_wdi = pd.DataFrame(df_wdi, dtype=np.float64)
+    df_wdi = df_wdi.reset_index()
+    df_wdi = df_wdi.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Select GDP and GDP pc in international-$ in 2017 prices
+    df_wdi = df_wdi[["country", "year", "ny_gdp_mktp_pp_kd", "ny_gdp_pcap_pp_kd"]]
+    df_wdi = df_wdi.rename(columns={"ny_gdp_mktp_pp_kd": "gdp", "ny_gdp_pcap_pp_kd": "gdp_per_capita"})
+    # Filter "World" entity
+    df_wdi = df_wdi[df_wdi["country"] == "World"]
+    # Drop empty World GDP estimations
+    df_wdi = df_wdi.dropna().reset_index(drop=True)
```

Comment:
I'm wondering if you may have nans only in one of the two data columns. If so, you could impose `how='all', subset=[...]`.

Reply:
Actually, it was because of the year (World estimations only go as far as 1990) and at the same time if GDP/GDP per capita estimations were missing. It's not the case here, but it happens sometimes with WB's regional aggregations